migrations/entitlements/migration.go

```diff
@@ -72,14 +87,30 @@ func ConvertToEntitledType(
 		return nil, fmt.Errorf("cannot migrate deprecated type: %s", staticType)
 	}
 
+	inter := mig.Interpreter
+	migratedTypeCache := mig.migratedTypeCache
+
+	staticTypeID := staticType.ID()
+
+	if migratedType, exists := migratedTypeCache[staticTypeID]; exists {
+		return migratedType, nil
+	}
+
+	defer func() {
+		if resultType != nil && conversionErr == nil {
+			migratedTypeCache[staticTypeID] = resultType
+		}
+	}()
```

Comment:
The cache might get accessed concurrently (cache is shared between account migrations, accounts are migrated concurrently). Maybe Make this a `sync.Map` and wrap it in a "`StaticTypeCache`"